docs/content/querying/lookups.md

```diff
@@ -0,0 +1,191 @@
+
+*example input*
+ 
+```json
+{"foo": "bar"}
+{"baz": "bat"}
+{"buck": "truck"}
+```
+
+*example lookupParseSpec*
+```json
+"lookupParseSpec":{
+  "type": "simpleJson"
+}
+```
+
+
+## JDBC lookup update
+
+The JDBC lookups will pull from a database cache. If the `tsColumn` is set it must be able to accept comparisons in the format `'2015-01-01 00:00:00'`. For example, the following must be valid sql for the table `SELECT * FROM some_lookup_table WHERE timestamp_column >  '2015-01-01 00:00:00'`. If `tsColumn` is set, the caching service will attempt to only pull values that were written *after* the last sync. If `tsColumn` is not set, the entire table is pulled every time.
+
+```json
+{
+  "type":"jdbc",
+  "lookup":"some_lookup",
+  "connectorConfig":{
+    "createTables":true,
+    "connectURI":"jdbc:mysql://localhost:3306/druid",
+    "user":"druid",
+    "password":"diurd"
+  },
+  "table":"some_lookup_table",
+  "keyColumn":"the_old_dim_value",
+  "valueColumn":"the_new_dim_value",
+  "tsColumn":"timestamp_column",
+  "updateMs":600000
+}
+```
+
+# Kafka lookup update (Experimental)
+It is possible to plug into a kafka topic whose key is the old value and message is the desired new value (both in UTF-8). This requires the following extension: "io.druid.extensions:druid-dim-rename-kafka8"
+```json
+{
+  "type":"kafka",
+  "lookup":"testTopic",
+  "kafkaTopic":"testTopic"
+}
+```
+## Kafka renames
+The extension `druid-dim-rename-kafka8` enables reading from a kafka feed which has name/key pairs to allow renaming of dimension values. An example use case would be to rename an ID to a human readable format.
+
+Currently the historical node caches the key/value pairs from the kafka feed in an ephemeral memory mapped DB via MapDB.
+
+## Configuration
+The following options are used to define the behavior:
+
+|Property|Description|Default|
+|--------|-----------|-------|
+|`druid.query.rename.kafka.properties`|A json map of kafka consumer properties. See below for special properties.|See below|
+
+The following are the handling for kafka consumer properties in `druid.query.rename.kafka.properties`
+
+|Property|Description|Default|
+|--------|-----------|-------|
+|`zookeeper.connect`|Zookeeper connection string|`localhost:2181/kafka`|
+|`group.id`|Group ID, auto-assigned for publish-subscribe model and cannot be overridden|`UUID.randomUUID().toString()`|
+|`auto.offset.reset`|Setting to get the entire kafka rename stream. Cannot be overridden|`smallest`|
+
+## Testing the kafka rename functionality
+To test this setup, you can send key/value pairs to a kafka stream via the following producer console
+`./bin/kafka-console-producer.sh --property parse.key=true --property key.separator="->" --broker-list localhost:9092 --topic testTopic`
+Renames can then be published as `OLD_VAL->NEW_VAL` followed by newline (enter)
```

Comment:
how do you feel about writing a tutorial for QTL upon release? I think it'll help out a lot of folks